### DISCUSSION TITLE
Pre-roll and post-roll times configuration

### DISCUSSION THREAD

**Discussion by vazquezjm** [blakeblackshear/frigate#8976] [Ask A Question] [answered]
Are pre-roll and post-roll times configurable? 

Based on [this](https://docs.frigate.video/configuration/record/#what-do-the-different-retain-modes-mean), I'd like to understand if pre-roll and post-roll times are configurable. IE: 10 seconds before motion is detected, and 20 seconds after motion stopped.

**Comment by NickM-27** [comment, marked as answer]
it depends on your configuration. There are `pre_capture` and `post_capture` options available but they still only get saved depending on the retention mode. 

So for example: if you have retention mode of motion and pre_capture of 10 seconds. a person walks out the front door and there was no motion on the camera before that time then there will not be the pre_capture because the 10 seconds before there was no motion. 

However if the same scenario happened except before the person went outside there was motion from something on the camera then that person recording will keep the 10 seconds before they walked outside.

**Reply by vazquezjm** [reply]
Thanks for your reply!
